Frontend/CO.CDP.OrganisationApp.Tests/Pages/ShareInformation/ShareCodeDetailsTests.cs

```diff
@@ -103,19 +104,6 @@ public async Task OnGetDownload_ShouldReturnFileResult_WhenPdfIsDownloadedSucces
         fileResult.FileDownloadName.Should().Be("HDJ2123F.pdf");
     }
 
-    [Fact]
-    public async Task OnGetDownload_ShouldRedirectToPageNotFound_WhenPdfResponseIsNull()
-    {
-        var shareCode = "HDJ2123F";
-
-        _dataSharingApiClientMock.Setup(x => x.GetSharedDataFileAsync(shareCode))
-            .ReturnsAsync((WebApiClient.FileResponse?)null);
-
-        var result = await _pageModel.OnGetDownload(shareCode);
-
-        result.Should().BeOfType<RedirectResult>().Which.Url.Should().Be("/page-not-found");
-    }
-
```

Comment:
Instead of removing this test, shouldn't there be a test where `GetSharedDataFileAsync` throws an exception?

And looking at the code in `CO.CDP.OrganisationApp\Pages\ShareInformation\ShareCodeDetails.cshtml.cs` - I wonder if that is wrong? It's testing for a null response. Shouldn't this be a try / catch?

```
var fileResponse = await dataSharingClient.GetSharedDataFileAsync(shareCode);

if (fileResponse == null)
{
    return Redirect("/page-not-found");
}
```